### PR TITLE
[et] Fix issues in publish command after the first full publish

### PR DESCRIPTION
**Pull request by tsapeta**
# Why

Fixing issues in publish command that we encountered so far.

# How

- Replaced `pod update <podNames> --no-repo-update` with `pod install --no-repo-update` – it's been causing updates of indirect dependencies.
- Filter changed files using npm-packlist and change the condition answering whether there are any unpublished changes from number of commits to number of modified "packlisted" files.
- Took into account new features when resolving suggested release type (at least minor) – it implies better support for non-native packages.
- Always use `gitHead` from package view as we already published all packages with the new command (there is a difference in what `gitHead` meant in the old script and what it means in the new one).
- Reverse commits order when printing package changes – it makes more sense imho.
- Put new line after empty version paragraph.
- Fix gathering workspaces info on newer Yarn versions.
- Use local version from package.json if that version hasn't been published yet, no matter of the changes (kind of a way to force specific version to be published, but it also fixes an issue when version of unpublished package was bumped according to changelog entries).

# Test Plan

Tried to run publish of a few packages using both `--dry` and `-l` flags.

### CODE REVIEW

File: tools/expotools/src/Packages.ts [hunk 1]
```diff
@@ -239,6 +239,19 @@ export class Package {
   async hasChangelogAsync(): Promise<boolean> {
     return fs.pathExists(this.changelogPath);
   }
+
+  /**
+   * Checks whether package has any native code (iOS, Android, C++).
+   */
+  async isNativeModuleAsync(): Promise<boolean> {
+    const dirs = ['ios', 'android', 'cpp'].map((dir) => path.join(this.path, dir));
```
Comment (sjchmiela) [inline tools/expotools/src/Packages.ts:247]: Weren't we supposed to fetch directories from `unimodule.json` somehow?

File: tools/expotools/src/publish-packages/tasks/updateIosProjects.ts [hunk 2]
```diff
@@ -39,13 +39,9 @@ export const updateIosProjects = new Task<TaskArgs>(
...
 
-        logger.log(
-          '  ',
-          `${green(nativeApp.packageName)}: updating`,
-          podspecNames.map((podspecName) => green(podspecName!)).join(', ')
-        );
+        logger.log('  ', `${green(nativeApp.packageName)}: Reinstalling pods...`);
 
-        await spawnAsync('pod', ['update', ...podspecNames, '--no-repo-update'], {
+        await spawnAsync('pod', ['install', '--no-repo-update'], {
```
Comment (sjchmiela) [inline tools/expotools/src/publish-packages/tasks/updateIosProjects.ts:44]: What we could do to make it even more fool-proof would be to update iOS projects **before** publishing/making any changes — if the command would make any changes in the directories that would mean that the state is not clean and in order to make the _Publish packages_ commit clean and single-purposed one would need to first update Pods directories by reinstalling pods and only afterwards return to publishing packages.

**Review by sjchmiela — APPROVED** [review]
Looks good to me, added two nice comments 💯

File: tools/expotools/src/Packages.ts (hunk 1 above)
Comment (tsapeta) [reply tools/expotools/src/Packages.ts:247]: Yeah, we were, however 
- we use `subdirectory` from `unimodule.json` only in `expo-gl-cpp` where the native Android code lives in `cpp` dir
- `unimodule.json` doesn't exist in `expo-image` and I guess any new package might follow this (or not, dunno ¯\_(ツ)_/¯ )
- isn't it easier to maintain? we don't use anything else in the monorepo and we don't plan afaik

File: tools/expotools/src/publish-packages/tasks/updateIosProjects.ts (hunk 2 above)
Comment (tsapeta) [reply tools/expotools/src/publish-packages/tasks/updateIosProjects.ts:44]: Interesting idea 🤔 I'm only afraid that installing pods is the bottleneck in terms of the performance, it takes the most time of the entire command, so running it twice would be quite annoying 😞

**Pull request merged**